measure_extinction/extdata.py

```diff
@@ -423,15 +423,41 @@ def calc_AV(self, akav=0.112):
 
         # if no SpeX spectrum is available: compute A(V) from E(K-V)
         else:
-            dwaves = np.absolute(self.waves["BAND"] - 2.19 * u.micron)
-            sindxs = np.argsort(dwaves)
-            kindx = sindxs[0]
-            if dwaves[kindx] > 0.02 * u.micron:
-                warnings.warn("no K band measurement in E(lambda-V)", UserWarning)
+            kindx = np.where(self.waves["BAND"] == 2.19 * u.micron)[0][0]
+            if not kindx:
+                warnings.warn(
+                    "No K band measurement available in E(lambda-V)!", stacklevel=2
+                )
             else:
                 ekv = self.exts["BAND"][kindx]
                 self.columns["AV"] = ekv / (akav - 1)
 
+    def calc_RV(self):
+        """
+        Calculate R(V) from the observed extinction curve
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        Updates self.columns["RV"]
+        """
+        # obtain or calculate A(V)
+        if "AV" not in self.columns.keys():
+            self.calc_AV()
+        av = self.columns["AV"]
+
+        # obtain E(B-V)
+        bindx = np.where(self.waves["BAND"] == 0.438 * u.micron)[0][0]
```

Comment:
Again, exact wavelength.  Is that necessary or is something close to this wavelength ok?

Reply:
I added some more flexibility now.